src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,354 @@
+                    object_reader=reader,
+                    embedding=embedding,
+                    config=config,
+                    environment_variables=environment_variables,
+                    **index_creation_kwargs,
+                )
+
+        index.update_index(
+            index_timestamp=index_timestamp,
+            workers=workers,
+            worker_resources=worker_resources,
+            worker_image=worker_image,
+            extra_worker_modules=extra_worker_modules,
+            driver_resources=driver_resources,
+            driver_image=driver_image,
+            extra_driver_modules=extra_driver_modules,
+            worker_access_credentials_name=acn,
+            max_tasks_per_stage=max_tasks_per_stage,
+            embeddings_generation_mode=embeddings_generation_mode,
+            embeddings_generation_driver_mode=embeddings_generation_driver_mode,
+            vector_indexing_mode=vector_indexing_mode,
+            config=config,
+            environment_variables=environment_variables,
+            namespace=namespace,
+            verbose=verbose,
+            trace_id=trace_id,
+            **index_update_kwargs,
+        )
+
+    graph = dag.DAG(
+        name="file-vector-search-ingestion",
+        mode=dag.Mode.BATCH,
+        max_workers=1,
+        namespace=namespace,
+    )
+    graph.submit(
+        ingest_files_udf,
+        file_dir_uri,
+        index_uri,
+        name="file-vector-search-ingestion",
+        access_credentials_name=acn,
+        resources=driver_resources,
+        image_name="vectorsearch",
+        acn=acn,
+        config=config,
+        environment_variables=environment_variables,
+        namespace=namespace,
+        verbose=verbose,
+        trace_id=trace_id,
+        create_index=create_index,
+        index_type=index_type,
+        index_creation_kwargs=index_creation_kwargs,
+        file_name=file_name,
+        include=include,
+        exclude=exclude,
+        suffixes=suffixes,
+        max_files=max_files,
+        text_splitter=text_splitter,
+        text_splitter_kwargs=text_splitter_kwargs,
+        embedding_class=embedding_class,
+        embedding_kwargs=embedding_class,
+        openai_key=openai_key,
+        index_timestamp=index_timestamp,
+        workers=workers,
+        worker_resources=worker_resources,
+        worker_image=worker_image,
+        extra_worker_modules=extra_worker_modules,
+        driver_resources=driver_resources,
+        driver_image=driver_image,
+        extra_driver_modules=extra_driver_modules,
+        max_tasks_per_stage=max_tasks_per_stage,
+        embeddings_generation_mode=embeddings_generation_mode,
+        embeddings_generation_driver_mode=embeddings_generation_driver_mode,
+        vector_indexing_mode=vector_indexing_mode,
+        index_update_kwargs=index_update_kwargs,
+    )
+    graph.compute()
+    graph.wait()
+
+
+ingest_files = as_batch(ingest_files_dag)
```

Comment:
This could be named `ingest` to follow the naming convention of the VCF implementation as a guide.

Reply:
Applied